kde-misc/kdialog/kdialog-22.12.2.recipe

```diff
@@ -0,0 +1,118 @@
+	lib:libKF5DBusAddons$secondaryArchSuffix
+	lib:libKF5GuiAddons$secondaryArchSuffix
+	lib:libKF5Codecs$secondaryArchSuffix
+	lib:libKF5Completion$secondaryArchSuffix
+	lib:libKF5I18n$secondaryArchSuffix
+	lib:libKF5IconThemes$secondaryArchSuffix
+	lib:libKF5ConfigCore$secondaryArchSuffix
+	lib:libKF5ConfigWidgets$secondaryArchSuffix
+	lib:libKF5Notifications$secondaryArchSuffix
+	lib:libKF5Solid$secondaryArchSuffix
+	lib:libKF5SonnetCore$secondaryArchSuffix
+	lib:libKF5TextWidgets$secondaryArchSuffix
+	lib:libKF5CoreAddons$secondaryArchSuffix
+	lib:libKF5ItemViews$secondaryArchSuffix
+	lib:libKF5JobWidgets$secondaryArchSuffix
+	lib:libKF5KIOCore$secondaryArchSuffix
+	lib:libKF5Service$secondaryArchSuffix
+	lib:libKF5WidgetsAddons$secondaryArchSuffix
+	lib:libKF5WindowSystem$secondaryArchSuffix
+	lib:libKF5XmlGui$secondaryArchSuffix
+	lib:libQt5Core$secondaryArchSuffix
+	lib:libQt5Gui$secondaryArchSuffix
+	lib:libQt5PrintSupport$secondaryArchSuffix
+	lib:libQt5Widgets$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	"
+
+SUPPLEMENTS="
+	kio_extras$secondaryArchSuffix
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	extra_cmake_modules$secondaryArchSuffix >= 5.77
+	devel:libKF5Auth$secondaryArchSuffix
+	devel:libKF5Bookmarks$secondaryArchSuffix
+	devel:libKF5DBusAddons$secondaryArchSuffix
+	devel:libKF5GuiAddons$secondaryArchSuffix
+	devel:libKF5Codecs$secondaryArchSuffix
+    devel:libKF5Completion$secondaryArchSuffix
+	devel:libKF5I18n$secondaryArchSuffix
+	devel:libKF5IconThemes$secondaryArchSuffix
+	devel:libKF5ConfigCore$secondaryArchSuffix
+	devel:libKF5ConfigWidgets$secondaryArchSuffix
+	devel:libKF5Notifications$secondaryArchSuffix
+	devel:libKF5Solid$secondaryArchSuffix
+	devel:libKF5SonnetCore$secondaryArchSuffix
+	devel:libKF5TextWidgets$secondaryArchSuffix
+	devel:libKF5CoreAddons$secondaryArchSuffix
+	devel:libKF5ItemViews$secondaryArchSuffix
+	devel:libKF5JobWidgets$secondaryArchSuffix
+	devel:libKF5KIOCore$secondaryArchSuffix
+	devel:libKF5Service$secondaryArchSuffix
+	devel:libKF5WidgetsAddons$secondaryArchSuffix
+	devel:libKF5WindowSystem$secondaryArchSuffix
+	devel:libKF5XmlGui$secondaryArchSuffix
+	devel:libQt5Core$secondaryArchSuffix
+	devel:libQt5Gui$secondaryArchSuffix
+	devel:libQt5PrintSupport$secondaryArchSuffix
+	devel:libQt5Widgets$secondaryArchSuffix
+	devel:libz$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	cmd:cmake
+	cmd:g++$secondaryArchSuffix
+	cmd:make
+	cmd:qdbuscpp2xml$secondaryArchSuffix >= 5
+	"
+
+BUILD()
+{
+	sed -e '/add_subdirectory(doc)/ s/^#*/#/' -i CMakeLists.txt
+	sed -e '/kdoctools_install(po)/ s/^#*/#/' -i CMakeLists.txt
+
+	mkdir -p build
+	cd build
+	cmake .. \
+		-DCMAKE_BUILD_TYPE=Release \
+		-DCMAKE_INSTALL_PREFIX=$prefix \
+		-DCMAKE_INSTALL_BINDIR=$commandBinDir \
+		-DSHARE_INSTALL_PREFIX=$dataDir \
+		-DECM_DIR=/system/data/cmake/Modules/ECM/cmake \
+		-DKDE_INSTALL_PLUGINDIR=$addOnsDir/Qt5
+
+	make $jobArgs
+}
+
+INSTALL()
+{
+	cd build
+	make install
```

Comment:
```suggestion

	cmake -S . -B build \
		-DCMAKE_BUILD_TYPE=Release \
		-DCMAKE_INSTALL_PREFIX=$prefix \
		-DCMAKE_INSTALL_BINDIR=$commandBinDir \
		-DSHARE_INSTALL_PREFIX=$dataDir \
		-DECM_DIR=/system/data/cmake/Modules/ECM/cmake \
		-DKDE_INSTALL_PLUGINDIR=$addOnsDir/Qt5

	make -C build $jobArgs
}

INSTALL()
{
	make -C build install
```

Reply:
It's preferred, but not required, to avoid the '`mkdir`/`cd` build' commands and use the tools themselves to change directories.